django/thunderstore/api/cyberstorm/views/community.py

```diff
@@ -0,0 +1,214 @@
+    )
+    def get(
+        self,
+        request: HttpRequest,
+    ) -> HttpResponse:
+
+        qp = CommunitiesQueryParameterCyberstormSerializer(data=request.query_params)
+        if not qp.is_valid():
+            return Response(qp.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        params: OrderedDict = qp.validated_data
+        communities_qs = self.get_queryset(params)
+        communities_qs = self.order_queryset(params["ordering"], communities_qs)
+        communities_page = self.paginate(params, communities_qs, params["page_size"])
+        communities_count = len(communities_qs)
+        serializer = self.serialize_results(
+            communities_page, communities_count, params["page_size"]
+        )
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def get_full_cache_vary(self, params: OrderedDict) -> str:
+        cache_vary = "communities_cache_key"
+        cache_vary += f".{params['ordering']}"
+        cache_vary += f".{params['page']}"
+        cache_vary += f".{params['page_size']}"
+        return cache_vary
+
+    def get_queryset(self, params=None) -> QuerySet[Community]:
+        return Community.objects.filter(is_listed=True,).annotate(
+            pkgs=Count(
+                "package_listings",
+                filter=Q(
+                    Q(package_listings__package__is_deprecated=False)
+                    & Q(
+                        Case(
+                            When(
+                                require_package_listing_approval=True,
+                                then=Case(
+                                    When(
+                                        Q(
+                                            package_listings__review_status=PackageListingReviewStatus.approved
+                                        ),
+                                        then=True,
+                                    ),
+                                    default=False,
+                                    output_field=BooleanField(),
+                                ),
+                            ),
+                            default=Case(
+                                When(
+                                    Q(
+                                        package_listings__review_status=PackageListingReviewStatus.unreviewed
+                                    )
+                                    | Q(
+                                        package_listings__review_status=PackageListingReviewStatus.approved
+                                    ),
+                                    then=True,
+                                ),
+                                default=False,
+                                output_field=BooleanField(),
+                            ),
+                        ),
+                    ),
+                ),
+                distinct=True,
+            ),
+            downloads=Sum(
+                Case(
```

Comment:
Are you certain it makes sense to use different "filters" when counting the downloads vs. when counting the packages?

Reply:
Hmm, yeah. I thought this downloads query was enough, but now that I look at it again, it's missing the unreviewed and approved checks. I'll fix this

Reply:
I'm not really convinced this is something we should be querying in real time in the first place. It would make more sense to aggregate per-community stats as a background job over some time cycle.

Since we don't have those fields yet, it'd be fine to abstract them behind functions that run a query per community, or just use stub values for now. That can then be replaced by the background job implementation in a separate PR.

Reply:
This has been solved